src/slowmokit/methods/preprocessing/one_hot_encoder.cpp

```diff
@@ -0,0 +1,29 @@
+#include "one_hot_encoder.hpp"
+
+template<class T>
+
+// Function to one hot encode a vector of integers
+std::vector<std::vector<int>> one_hot_encoder(const std::vector<int> &data, int n_classes) {
+  // Created a map to store the mapping of each integer in the data to a different index
+  std::map<int, int> class_mapping;
+  int class_index = 0;
+  // added each integer to class mapping
+  for (int value : data) {
+    if (class_mapping.count(value) == 0) {
+      class_mapping[value] = class_index++;
+    }
+  }
+
+  // Vector to store the one hot encoded data
+  std::vector<std::vector<int>> one_hot_encoded_data;
```

Comment:
Declare the vector size here itself, pushback might be slower

Reply:
done